includes/class-wc-gateway-swedbank-pay-checkout.php

```diff
@@ -0,0 +1,1837 @@
+			$shipping        = $order->get_shipping_total();
+			$tax             = $order->get_shipping_tax();
+			$shippingWithTax = $shipping + $tax;
+			$taxPercent      = ( $tax > 0 ) ? round( 100 / ( $shipping / $tax ) ) : 0;
+			$shippingMethod = trim( $order->get_shipping_method() );
+
+			$items[] = [
+				'reference' => 'shipping',
+				'name' => ! empty( $shippingMethod ) ? $shippingMethod : __( 'Shipping', 'woocommerce' ),
+				'type' => 'SHIPPING_FEE',
+				'class' => 'ProductGroup1',
+				'quantity' => 1,
+				'quantityUnit' => 'pcs',
+				'unitPrice' => round( $shippingWithTax * 100 ),
+				'vatPercent' => round( $taxPercent * 100 ),
+				'amount' => round( $shippingWithTax * 100 ),
+				'vatAmount' => round( $tax * 100 )
+			];
+		}
+
+		// Add fee lines
+		foreach ( $order->get_fees() as $order_fee ) {
+			/** @var WC_Order_Item_Fee $order_fee */
+			$fee        = $order_fee->get_total();
+			$tax        = $order_fee->get_total_tax();
+			$feeWithTax = $fee + $tax;
+			$taxPercent = ( $tax > 0 ) ? round( 100 / ( $fee / $tax ) ) : 0;
+
+			$items[] = [
+				'reference' => 'fee',
+				'name' => $order_fee->get_name(),
+				'type' => 'OTHER',
+				'class' => 'ProductGroup1',
+				'quantity' => 1,
+				'quantityUnit' => 'pcs',
+				'unitPrice' => round( $feeWithTax * 100 ),
+				'vatPercent' => round( $taxPercent * 100 ),
+				'amount' => round( $feeWithTax * 100 ),
+				'vatAmount' => round( $tax * 100 )
+			];
+		}
+
+		// Add discount line
+		if ( $order->get_total_discount( false ) > 0 ) {
+			$discount        = abs( $order->get_total_discount( true ) );
+			$discountWithTax = abs( $order->get_total_discount( false ) );
+			$tax             = $discountWithTax - $discount;
+			$taxPercent      = ( $tax > 0 ) ? round( 100 / ( $discount / $tax ) ) : 0;
+
+			$items[] = [
+				'reference' => 'discount',
+				'name' => __( 'Discount', WC_Swedbank_Pay_Checkout::TEXT_DOMAIN ),
+				'type' => 'DISCOUNT',
+				'class' => 'ProductGroup1',
+				'quantity' => 1,
+				'quantityUnit' => 'pcs',
+				'unitPrice' => round( - 100 * $discountWithTax ),
+				'vatPercent' => round( 100 * $taxPercent ),
+				'amount' => round( - 100 * $discountWithTax ),
+				'vatAmount' => round( - 100 * $tax )
+			];
+		}
+
+		return $items;
+	}
+
+	/**
+	 * Get Order VAT
+	 *
+	 * @param int $vatAmount
+	 * @param WC_Order $order
+	 *
+	 * @return int
+	 */
+	public static function checkout_order_vat( $vatAmount, $order ) {
+		$items = apply_filters( 'swedbank_pay_checkout_order_items', [], $order );
+
+		return array_sum( array_column( $items, 'vatAmount') );
+	}
+
+	/**
+	 * Get Order Amount
+	 *
+	 * @param int $amount
+	 * @param WC_Order $order
+	 *
+	 * @return int
+	 */
+	public static function checkout_order_amount( $amount, $order ) {
+		return round( 100 * $order->get_total() );
+	}
+
+	/**
+	 * Get Consumer Profile
+	 *
+	 * @param $user_id
+	 *
+	 * @return array
+	 */
+	protected function get_consumer_profile( $user_id ) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The method get_consumer_profile() has a Cyclomatic Complexity of 14. The configured cyclomatic complexity threshold is 10.](https://app.codacy.com/gh/SwedbankPay/swedbank-pay-woocommerce-payments/pullRequest?prid=5340032)